apps/mastodoncounter/mastodon_counter.star

```diff
@@ -0,0 +1,222 @@
+                                src = MASTODON_ICON,
+                            ),
+                            render.WrappedText("MASTODON FOLLOWERS"),
+                        ],  #children
+                    ),  #render.Row
+                ),  #render.Box
+                render.Box(
+                    width = 64,
+                    height = 14,
+                    color = "#000000",
+                    child = render.Row(
+                        expanded = True,
+                        children = [
+                            render.Marquee(
+                                width = 64,
+                                offset_start = 64,
+                                child = render.Text(
+                                    font = "6x13",
+                                    content = "Something's up with your Mastodon instance or API key. Check your settings in the Tidbyt mobile app.",
+                                ),
+                            ),
+                        ],  #children
+                    ),  #render.Row
+                ),  #render.Box
+            ],  #children
+        ),  #render.Column
+    )  #render.Root
+
+def connectionError():
+    return render.Root(
+        delay = 0,
+        child = render.Column(
+            children = [
+                render.Box(
+                    width = 64,
+                    height = 18,
+                    color = "#6364ff",
+                    child = render.Row(
+                        expanded = True,
+                        children = [
+                            render.Image(
+                                width = 18,
+                                height = 16,
+                                src = MASTODON_ICON,
+                            ),
+                            render.WrappedText("MASTODON FOLLOWERS"),
+                        ],  #children
+                    ),  #render.Row
+                ),  #render.Box
+                render.Box(
+                    width = 64,
+                    height = 14,
+                    color = "#000000",
+                    child = render.Row(
+                        expanded = True,
+                        children = [
+                            render.Marquee(
+                                width = 64,
+                                offset_start = 64,
+                                child = render.Text(
+                                    font = "6x13",
+                                    content = "We couldn't reach Mastodon. If you're sure your settings are correct, your instance might be having issues.",
+                                ),
+                            ),
+                        ],  #children
+                    ),  #render.Row
+                ),  #render.Box
+            ],  #children
+        ),  #render.Column
+    )  #render.Root
+
+def main(config):
+    theapi = "https://%s/api/v1/accounts/verify_credentials" % config.str("instance", DEFAULT_SERVER)
```

Comment:
We can unfortunately not merge apps that perform http calls to arbitrary hosts. This is so to protect against our backend being abused to spam or DoS others.

Would it be possible to replace the `instance` field with perhaps a dropdown of popular mastodon instances instead?

Reply:
There are apparently 13,394 Mastodon instances right now, so... probably not? 😅 Not without having to exclude a bunch of people and/or do _constant_ support haha.

Your other two suggestions seem perfectly do-able but I can't think of a good way of solving this one. I'll mull it over for a while.

Reply:
Sorry for the back and forth @meejle! One thing that might be useful is https://instances.social/. They have an API here: https://instances.social/api/doc/. It might help get around the problem of arbitrary URLs

Reply:
@meejle I did integrate with instances.social in this PR: https://github.com/tidbyt/community/pull/972